ner_v2/tests/numeral/number/en/test_number_detection.py

```diff
@@ -1,254 +1,84 @@
-        """
-        message = 'I bought a car toy for 2.3k rupees'
-        number_detector_object = NumberDetector(entity_name=self.entity_name, language='en')
-        number_dicts, original_texts = number_detector_object.detect_entity(message)
-
-        zipped = list(zip(number_dicts, original_texts))
-        self.assertEqual(len(zipped), 1)
-        self.assertIn(({'value': '2300', 'unit': None}, u'2.3k'), zipped)
-
-    def test_en_number_detection_for_decimal_number_with_scale_and_unit_and_unit_type_given(self):
-        """
-        Number detection for english language for decimal number with scale like '1.2 thousand rupees', 'Rupees 2.2k'
-        """
-        message = 'I bought a car toy for 2.3k rupees'
-        number_detector_object = NumberDetector(entity_name=self.entity_name, language='en', unit_type='currency')
-        number_dicts, original_texts = number_detector_object.detect_entity(message)
-
-        zipped = zip(number_dicts, original_texts)
-        self.assertEqual(len(zipped), 1)
-        self.assertIn(({'value': '2300', 'unit': 'rupees'}, u'2.3k rupees'), zipped)
-
-    def test_en_number_detection_for_decimal_number_with_scale_and_unit_and_different_unit_type_given(self):
-        """
-        Number detection for english language for decimal number with scale like '1.2 thousand rupees', 'Rupees 2.2k'
-        """
-        message = 'I buys 2.3k kg mango'
-        number_detector_object = NumberDetector(entity_name=self.entity_name, language='en', unit_type='currency')
-        number_dicts, original_texts = number_detector_object.detect_entity(message)
-
-        zipped = list(zip(number_dicts, original_texts))
-        self.assertEqual(len(zipped), 0)
+
+
+class NumberDetectorTestsMeta(type):
+    yaml_test_files = [
+        os.path.join(os.path.dirname(os.path.abspath(__file__)), "number_ner_tests.yaml")
+    ]
+
+    def __new__(cls, name, bases, attrs):
+        for test_name, test_fn in cls.yaml_testsuite_generator():
+            attrs[test_name] = test_fn
+
+        return super(NumberDetectorTestsMeta, cls).__new__(cls, name, bases, attrs)
+
+    @classmethod
+    def yaml_testsuite_generator(cls):
+        for filepath in cls.yaml_test_files:
+            test_data = yaml.load(io.open(filepath, "r", encoding="utf-8"))
+            for language in test_data["tests"]:
+                for i, testcase in enumerate(test_data["tests"][language]):
+                    yield (
+                        "test_yaml_{}".format(testcase["_id_"]),
+                        cls.get_yaml_test(testcase=testcase,
+                                          language=language,)
+                    )
+
+    @classmethod
+    def get_yaml_test(cls, testcase, language, **kwargs):
+        def parse_expected_outputs(expected_outputs):
+            num_dicts, original_texts = [], []
+            for expected_output in expected_outputs:
+                if "detected_number" in expected_output:
```

Comment:
Why is this needed? I am not sure where this applies

Reply:
This check is for those commented test cases.